capa/ghidra/capa_explorer.py

```diff
@@ -0,0 +1,261 @@
+
+            if to_cont:
+                continue
+
+            # greedily create labels at basic block & insn scopes
+            node_to_parse = self.node[self.locations.index(addr)]
+            if node_to_parse:
+                txt = self.recurse_node(node_to_parse)
+
+                if not txt:
+                    continue
+
+                txt = txt.replace(" ", "-")
+                createLabel(a, txt, name_space, True, SourceType.USER_DEFINED)  # type: ignore [name-defined] # noqa: F821
+                continue
+
+            # handle first address of match
+            createLabel(a, txt, name_space, True, SourceType.USER_DEFINED)  # type: ignore [name-defined] # noqa: F821
+
+
+def get_capabilities():
+    rules_dir: str = ""
+    try:
+        selected_dir = askDirectory("Choose capa rules directory", "Ok")  # type: ignore [name-defined] # noqa: F821
+        if selected_dir:
+            rules_dir = selected_dir.getPath()
+    except RuntimeError:
+        # RuntimeError thrown when user selects "Cancel"
+        pass
+
+    if not rules_dir:
+        logger.info("You must choose a capa rules directory before running capa.")
+        return capa.main.E_MISSING_RULES
+
+    rules_path: pathlib.Path = pathlib.Path(rules_dir)
+    logger.info("running capa using rules from %s", str(rules_path))
+
+    rules = capa.main.get_rules([rules_path])
+    meta = capa.ghidra.helpers.collect_metadata([rules_path])
+    extractor = capa.features.extractors.ghidra.extractor.GhidraFeatureExtractor()
+
+    capabilities, counts = capa.main.find_capabilities(rules, extractor, True)
+
+    if capa.main.has_file_limitation(rules, capabilities, is_standalone=False):
+        logger.info("capa encountered warnings during analysis")
+
+    return capa.render.json.render(meta, rules, capabilities)
+
+
+def get_locations(match_dict):
+    """recursively collect match addresses and associated nodes"""
+    if "locations" in match_dict:
+        for loc in match_dict.get("locations", {}):
+            if "type" in loc:
+                # either an rva (absolute)
+                # or an offset into a file (file)
+                if loc.get("type", "") in ("absolute", "file"):
+                    yield loc.get("value"), match_dict.get("node")
+
+    if match_dict["children"]:
+        for child in match_dict.get("children", {}):
+            yield from get_locations(child)
+
+
+def parse_json(capa_data):
+    """Parse json produced by capa"""
+    ghidra_data = []
+
+    rules = capa_data["rules"]
+    for rule in rules.keys():
+        # loosely coupled location & node data lists
+        this_locs = []
+        this_node: List[Dict] = []
+
+        # dict data of currently matched rule
+        this_capability = rules[rule]
+        meta = this_capability["meta"]
+
+        # get Mitre ATT&CK and MBC
+        this_attack = meta.get("attack")
+        this_mbc = meta.get("mbc")
```

Comment:
What happens if `this_attack` and/or `this_mbc` are set to `None` here?

Reply:
It will proceed with empty lists that contain dictionary data. Bookmarking will be skipped for empty ones.

So far these keys always exist and return the empty lists; however, I've added a check in the newest push